contents/tutorials/vue-cookie-banner.md

```diff
@@ -0,0 +1,298 @@
+---
+
+With internet privacy regulations, like GDPR, coming into effect, managing cookies is becoming increasingly important. Cookies are pieces of information apps set in users’ browsers to help them store information and identity. It’s possible to use [PostHog without cookies](/tutorials/cookieless-tracking), but it’s simpler to use them.
+
+To ensure you are compliant with regulations like GDPR, your app must receive consent to use cookies. The cookie consent banner is a way to do this, and this tutorial shows you how to build one in Vue, a popular JavaScript framework.
+
+## Step 1: Setting up our Vue app
+
+First, if you haven’t used Vue before, install the command line interface (CLI). You also need a package manager like npm, which you can get by installing [Node](https://nodejs.org/en/).
+
+```bash
+npm install -g @vue/cli
+```
+
+After installing the CLI, you can create the Vue project. We are naming ours "tutorial." When the CLI prompts you with options, choose `vue3` and `npm` (or yarn, it doesn’t matter).
+
+```bash
+vue create tutorial
+```
+
+Next, head into the folders it created and run the server. This launches the site. 
+
+```bash
+cd tutorial
+npm run serve
+```
+
+Now you have the template for a Vue app that is ready for PostHog and a cookie consent banner. 
+
+## Step 2: Installing PostHog
+
+We’ll add PostHog as a plugin to Vue. This way we can access it everywhere in our Vue app and use it to make our cookie banner functional. To do this, install `posthog-js`, create a `plugins` folder, and a `posthog.js` file within that folder.
+
+```bash
+npm i posthog-js
+cd src
+mkdir plugins
+touch posthog.js
+```
+
+In `posthog.js`, add the code to set up and initialize PostHog as a plugin.
+
+```js
+// src/plugins/posthog.js
+import posthog from "posthog-js";
+
+export default {
+  install(app, options) {
+    app.config.globalProperties.$posthog = posthog.init(
+      "<ph_project_api_key>",
+      {
+        api_host: "<ph_instance_address>",
+      }
+    );
+  },
+};
+```
+
+In `main.js`, import the PostHog plugin and connect it to our app.
+
+```js
+// main.js
+import { createApp } from 'vue'
+import App from './App.vue'
+import posthogPlugin from "./plugins/posthog"; //import the plugin. 
+
+const app = createApp(App);
+
+app.use(posthogPlugin); //install the plugin
+app.mount('#app')
+```
+
+Once you run `npm run serve` again and head into the cookies for your site, you should see a PostHog cookie added.
+
+![Cookie](../images/tutorials/vue-cookie-banner/cookie.png)
+
+Our app did this without the consent of the user, and now, it’s time to build the cookie consent banner to fix this.
+
+## Step 3: Creating our cookie banner component
+
+First, create a `CookieBanner.vue` component in our `components` folder.
+
+```bash
+cd components
+touch CookieBanner.vue
+```
+
+For our cookie banner, create a basic component with some details about cookies, buttons to accept and decline, and some basic styling.
```

Comment:
May be fun to reference the file we use in our repo here, so users can see what the messaging should be?

Reply:
Yup, added 😄